sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -879,6 +879,17 @@ class Column(val expr: Expression) extends Logging {
    */
   def getItem(key: Any): Column = withExpr { UnresolvedExtractValue(expr, Literal(key)) }
 
+  /**
+   * An expression that adds/replaces a field by name in a `StructType`.
+   * If schema contains multiple fields with fieldName, they will all be replaced with fieldValue.
+   *
+   * @group expr_ops
+   * @since 3.1.0
+   */
+  def withField(fieldName: String, fieldValue: Column): Column = withExpr {
```

Comment:
By definition, `A column is collection of cells aligned vertically in a table. A field is an element in which one piece of information is stored, such as the Received field`.

Wondering if we should rename `withField` to `withColumn`? In this case, we can have consistent API to `Dataset`.

Reply:
Personally, `Column.withColumn` doesn't make as much sense as `Column.withField`. Given that a `StructType` dataType consists of a collection of `StructFields`, `Column.withField` clearly communicates to me that this method operates only on `StructType` dataType columns. Additionally, `Column.withField` nicely complements the existing `Column.getField` method. 

I'm not saying I'm opposed to making this change. Frankly, both names have their merits. I am only slightly in favour of `Column.withField`.

Reply:
Fair!